VERSIONING.md

```diff
@@ -0,0 +1,198 @@
+  verification, but now it does not). However, "security features" should not
+  break backwards-compatibility.
+* Any "unstable" commands.
+
+Violations of this are generally considered bugs; we will issue patch releases
+for each affected minor version. In some (rare) cases, an exception may be made
+with the approval of the Cosign maintainers as long as the decision results in
+an update to this versioning policy.
+
+[hyrums-law]: https://www.hyrumslaw.com/
+
+### Unstable commands
+
+It’s important to enable rapid iteration on the development of new Cosign
+features while providing stability to users of more-mature features. As such,
+the Cosign CLI will support the following release levels with corresponding
+guarantees:
+
+* **Experimental**: No guarantees. May change in a backwards-incompatible way or
+  be removed at any time without warning.
+* Generally available (**GA**): follows the guidelines in this document.
+
+Users must explicitly indicate that they want to opt-in to experimental behavior
+by setting the environment variable `COSIGN_EXPERIMENTAL=1`.
+
+### Supported versions
+
+**Full support.** We fully support only the latest release of the Cosign CLI.
+
+**Security support.** We will backport security fixes to all major versions
+released in the past year. We may backport security fixes, depending on
+severity, to all minor versions released in the last year. We may choose to
+backport security "features" as well.
+
+### Deprecation
+
+Any behavior may be deprecated at any time with sufficient warning. For GA
+features, that period is 6 months from the Cosign release in which the
+deprecation was first announced; the first subsequent Cosign release should
+remove that feature. A new major version voids all deprecation timeline
+guarantees, and any deprecated behaviors can be removed at that time.
+
+Deprecation requires:
+
+* When users invoke the deprecated behavior, print a deprecation message to
+  standard error.
+  * The message should look something like:
+    > WARNING: $BEHAVIOR is deprecated and will be removed in a Cosign release
+    > soon after $DEPRECATION_DATE (see $GITHUB_ISSUE_LINK). Instead, please
+    > $ALTERNATIVE.
+  * If we can’t tell whether a user is using the deprecated behavior (for
+    instance, if we’re going to remove a field), print the message every time.
+  * There MAY be a way to turn off the message. In many cases, this will just be
+    by using the alternative. Otherwise, we can consider adding a flag (which is
+    now subject to a deprecation period of its own).
+* A release note at the introduction of the deprecation, along with the removal
+  of the behavior.
+* Any documentation that refers to the deprecated behavior in the Cosign
+  repository should have the same information as in the message.
+  * Other documentation will be updated on a best-effort basis.
+
+For a list of currently deprecated behavior, search the codebase.
```

Comment:
lulz "search the codebase"?

Reply:
This is actually a reasonable thing to do because we'll have a "deprecated behavior library" that all deprecated code imports. Added a note